lab/src/raft/leader_election.go

```diff
@@ -8,6 +8,21 @@ func (rf *Raft) SendHeartbeat() {
 	// send heartbeat to all other followers
 }
 
+func (rf *Raft) BackToFollower(newTerm int) {
+	rf.mu.Lock()
+	rf.currentTerm = newTerm
+	rf.votedFor = -1
+	if rf.status == candidate {
+		rf.closeVoteChan <- true
+	} else if rf.status == leader {
+		rf.clockInterval = time.Duration(rand.Intn(150) + 200 + (rf.me*100)%150)
+	}
+	rf.status = follower
+	rf.onEmptyElectionTerm = false
+	rf.SetHeatBeatClock()
+	rf.mu.Unlock()
+}
+
```

Comment:
Let's consider renaming `heartbeatClock` to `clock`.

Reply:
Done